src/coap_gnutls.c

```diff
@@ -722,9 +724,12 @@ static int cert_verify_gnutls(gnutls_session_t g_session)
   cert_type = get_san_or_cn(g_session, &cert_info);
 #if (GNUTLS_VERSION_NUMBER >= 0x030606)
   if (cert_type == GNUTLS_CRT_RAW)
-    goto finish;
+    goto check_verify;
```

Comment:
To avoid jumping back and forth via `goto` I suggest moving the contents after the `check_verify` label into a helper function that returns 1 on success and 0 in case of an error. In the latter case, you would `goto fail` as is done for the other (non-RPK) checks. On success, just `return 1`.

Reply:
Change made as suggested and code pushed.

Reply:
Valgrind is complaining about something - checking into it.

Reply:
Valgrind issue fixed. Caused by update_pki_key() free'ing off PEM_BUF memory used for initial setting up PKI for new incoming session.

Reply:
Code pushed